src/main/app/common/court-calculations/courtDetermination.ts

```diff
@@ -12,16 +12,25 @@ export class CourtDetermination {
                             claimantPaymentDate: Moment,
                             courtGeneratedPaymentDate: Moment): DecisionType {
 
-    if (!defendantPaymentDate || !claimantPaymentDate || !courtGeneratedPaymentDate) {
+    if (!defendantPaymentDate || !claimantPaymentDate) {
```

Comment:
As discussed, overall the logic can be simplified here. Something like this is easier to reason about:

```
import { Moment } from 'moment'

export enum DecisionType {
  CLAIMANT = 'CLAIMANT',
  DEFENDANT = 'DEFENDANT',
  COURT = 'COURT'
}

export class CourtDetermination {

  static calculateDecision (defendantPaymentDate: Moment,
                            claimantPaymentDate: Moment,
                            courtGeneratedPaymentDate: Moment): DecisionType {

    if (!defendantPaymentDate || !claimantPaymentDate) {
      throw new Error('Input should be a moment, cannot be empty')
    }

    if (claimantPaymentDate.isSameOrAfter(defendantPaymentDate)) {
      return DecisionType.CLAIMANT
    }

    if (!courtGeneratedPaymentDate) {
      return DecisionType.DEFENDANT
    }

    if (courtGeneratedPaymentDate.isBefore(claimantPaymentDate)) {
      return DecisionType.COURT
    }

    if (courtGeneratedPaymentDate.isBefore(defendantPaymentDate)) {
      return DecisionType.COURT
    }
    
    return DecisionType.DEFENDANT
  }
}
```